### DISCUSSION TITLE
Any idea why version has to be a number?

### DISCUSSION THREAD

**Discussion by reimertz** [pmndrs/zustand#2199] [Ideas] [answered]
Hi there! 

When setting up a new project, we accidentally started using a semver instead of a number for our zustand versioning. This would not have been a problem in a TS-based project since we'd get the errors.

But that got me thinking; Is there a reason why we don't allow it to be a string as well? Since the current check only check if `oldVersion != newVersion`, I don't see why we couldn't allow this feature.

**Comment by dai-shi** [comment, marked as answer]
Yeah, you are right. I'm not sure, but we may want to do `oldVersion < newVersion` in the future, so maybe just leaving it would be a safe bet.
You can hack with `as number` with locking zustand version, I guess.

**Reply by reimertz** [reply]
Thanks for the answer. It seems the js-compiled version of zustand/persist contains the `typeof number` check still, so not sure above solution would get around the issue.

**Reply by dai-shi** [reply]
Ah, you are right.

**Reply by dbritto-dev** [reply]
@reimertz you can try with `as never`
